misk/web/tabs/web-actions/src/ducks/webActions.ts

```diff
@@ -183,12 +200,14 @@ function* handleMetadata() {
           nonAccessOrTypeFunctionAnnotations
         }
       })
-      .groupBy("pathPattern")
+      .groupBy(groupByWebActionHash)
       .map((actions: IWebActionInternal[]) => {
-        const dispatchMechanism = flatMap(
-          actions,
-          action => action.dispatchMechanism
-        )
+        const dispatchMechanism = chain(actions)
+          .flatMap(action => action.dispatchMechanism)
```

Comment:
this should be handled server side, we should leave a TODO here to remove this duplicate processing